### ISSUE TITLE
Mobile app login sessions overwriting each other

### ISSUE THREAD

**Issue by pesintta** [mattermost/mattermost-server#9311]
#### Summary
Multiple mobile app (iOS?) sessions to same server are not working simultaneously

#### Steps to reproduce
1. Log-in to Mattermost with username & password using Mattermost app in iPhone
2. Leave the application logged in and use another iOS device to log-in with Mattermost app using the same credentials to the same server. I used iPad in my tests.
3. The second login succeeds. Now open the application on the first device again and observe that its session has been logged out for some reason.

This is experienced with Mattermost 5.2.0 Team Edition and latest version of the iOS application (1.11.0). Unfortunately I couldn't test the behavior with Android.

#### Expected behavior
It would be reasonable to assume that both devices will maintain their sessions until they expire or are manually logged-out.

#### Observed behavior (that appears unintentional) 
The device that was first logged in had its session "overwritten" by the second login.

There are also messages such as the following in mattermost server logs:
```
{"level":"info","ts":1535277344.4793856,"caller":"web/handlers.go:95","msg":"Invalid session","error":"GetSession: Invalid session token=t3wro4dxipbpfcbqcstqrs8ano, err=<no value>, "}
```

Here is a screenshot of the active sessions dialog from a browser while the iPhone is logged in:
![image](https://user-images.githubusercontent.com/9271441/44627134-37d50880-a931-11e8-8253-9c3139c3d054.png)

And here is the contents of the dialog after the second login was performed:
![image](https://user-images.githubusercontent.com/9271441/44627145-5cc97b80-a931-11e8-9807-82f2c52501fe.png)

As you can see the iPhone session has vanished from the list.

It seems that only the mobile application is affected. Using Safari instead of the Mattermost app on the second device does not lead to the first session expiring.

**Comment by pesintta** [comment]
I was able to do some more investigation today and I think I got this one figured out.

Looking at the database it seems that if push notifications are not permitted on application launch some versions of Mattermost on iOS ended up sending a deviceId with the string contents "null" to the server.
If two or more devices send a deviceId "null" they are considered the same device and their existing sessions are revoked when either one of them performs login.

Relevant code for that is here:
https://github.com/mattermost/mattermost-server/blob/v5.2.0/app/login.go#L135

As a fix it might be sensible to sanitize the deviceId in the app or on server side to treat string value of "null" the same as empty value "" which will then be taken care of by the mattermost server properly as seen here: https://github.com/mattermost/mattermost-server/blob/master/app/login.go#L132

Side note: Basing native mobile app detection on DeviceId being present seems like a suboptimal choice as many users opt out of push notifications in which case the detection fails as deviceId is then not present.
